src/main/java/com/example/petbuddybackend/utils/keycloak/KeycloakJwtAuthenticationConverter.java

```diff
@@ -0,0 +1,82 @@
+package com.example.petbuddybackend.utils.keycloak;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.security.authentication.AbstractAuthenticationToken;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.security.oauth2.jwt.JwtClaimNames;
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken;
+import org.springframework.security.oauth2.server.resource.authentication.JwtGrantedAuthoritiesConverter;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Configuration
+@Slf4j
+public class KeycloakJwtAuthenticationConverter implements Converter<Jwt, AbstractAuthenticationToken> {
+
+    private static final String RESOURCE_ACCESS = "resource_access";
+
+    @Value("${keycloak.resource-roles-claim}")
+    private String resourceRolesClaim;
+
+    private static final String ROLES = "roles";
+
+    @Value("${keycloak.claim-name}")
+    private String claimName;

```

Comment:
Plus widzę, że niżej sprawdzane jest czy ta wartość jest null żeby wsadzić parametr
domyślny. Jeżeli faktycznie przewidujemy żeby to niekoniecznie znajdowało się 
propertiesach to zamiast sprawdzać czy to jest null to dałbym default w ten sposób:

    @Value("${keycloak.claim-name:" + JwtClaimNames.SUB + "}")

Reply:
oo fajne, chociaz dziwne ze to moze byc null, nwm czy cos nie pomylilem wiec zerkne na to jeszcze